pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy

```diff
@@ -579,7 +579,7 @@ class ModelInterpreter implements Serializable {
         toolsList.each { l ->
             String k = l.get(0)
             Closure v = (Closure)l.get(1)
-            String toolVer = v.call()
+            String toolVer = delegateAndExecute(v)
```

Comment:
Added because of test failures.  Tried removing after latest changes, tests fail

This wasn't really required before this change because the `closure.owner` would be tried and `closure.owner` was the next outer closure which masked is bug. 
Now that the `owner` isn't the next closure out, delegate needs to be set properly.